test/viewer.js

```diff
@@ -128,6 +127,43 @@ test('get a message', t => {
 
 })
 
+test('get a message (publicWebHosting=false)', t => {
+    // now post a message by alice
+    sbot.db.publishAs(alice, {
+        type: 'post',
+        text: 'say hi'
+        }, (err, msg) => {
+        t.error(err, 'alice posts')
+
+        // publish a threaded response by user who has opted out of public web hosting
+        sbot.db.publishAs(dan, {
+            type: 'post',
+            text: 'hi',
+            root: msg.key
+        }, (err) => {
+            t.error(err, 'dan posts a threaded response')
+
+            // finally get the message
+            fetch(BASE_URL + '/msg/' + encodeURIComponent(msg.key))
+            .then(res => res.json())
+            .then(({ messages }) => {
+                const firstMsg = messages.find(m => msg.key === m.key)
+                t.true(firstMsg, 'should return alices post')
+    
+                // try find the threaded response that dan posted
+                const threadedMsg = messages.find(m => m.value.author === dan.id)
+                t.false(threadedMsg, 'should not return dans threaded response')
```

Comment:
IMO there should be two messages come back, but one of them has been hollowed out content + author wise